### PR TITLE
Rewrote monitor & consume events

### PR DESCRIPTION
**Pull request by andrzej-casper**
### What does this PR change?

Rewrote Build-dApps/Monitor-and-Consume-Events.

### Additional context

These changes are backported from the default branch of [docs-new](https://github.com/casper-network/docs-new) repository, so you can preview them [live here](https://staging.docs.casper.network/developers/dapps/monitor-and-consume-events/).

Original PR authored by @dylanireland: https://github.com/casper-network/docs-new/pull/181.

### Checklist

- [x] I ran the docs locally using `yarn install` and `yarn run start`.
- [x] All links (internal and external) have been verified.
- [x] All technical procedures have been tested.

### CODE REVIEW

**Review by ipopescu — CHANGES_REQUESTED** [review]
@dylanireland, thanks for the content!
I spoke about this PR with @ACStoneCL and @darthsiroftardis, and we'd like to keep the curl statements as well. Is it possible to use different tabs to keep the original content we had on the page?

**Comment by dylanireland** [comment]
cURL documentation re-added in multilingual code blocks / tabs. Requesting review:
@ipopescu 
@ACStoneCL 
@darthsiroftardis

**Comment by ipopescu** [comment]
To proceed with this PR:
- @dylanireland @andrzej-casper @ACStoneCL, please review PR #1108.
- @hoffmannjan, please review the technical code blocks or re-assign the review to someone on the team.

Also, note that much of this content will need to be replaced once the SSE sidecar is released. The sidecar is already running in production in beta mode with some validators. We will be discouraging users to connect to the node directly.

"The Casper Event Sidecar is an application that runs in tandem with the node process. This reduces the load on the node process by allowing subscribers to monitor the event stream through the Sidecar while the node focuses entirely on the blockchain."

Thank you!

**Comment by dylanireland** [comment]
LGTM. Requesting review from @hoffmannjan and @ACStoneCL

File: source/docs/casper/developers/dapps/monitor-and-consume-events.md [hunk 1]
```diff
@@ -1,664 +1,488 @@
...
 
-**Finality Signature event**
+:::
 
-This event indicates that the final approvals from validators are signed and further alterations to the block will not be allowed. Refer to the [consensus reached](../../concepts/design/casper-design.md#consensus-reached) section and [block finality](../../concepts/glossary/B.md#block-finality) 
-section to learn more about finality signatures.
+## Listening to the Event Stream
 
-**Main events**
+Applications can listen for such events for a specific account during a particular era, containing certain data. Then, they can parse the data and discard what is not needed. To consume the event stream, set up an event listener in your dApp using the following code:
```
Comment (ACStoneCL) [inline source/docs/casper/developers/dapps/monitor-and-consume-events.md:23]:
```suggestion
Applications can listen for such events for a specific account during a particular era, containing certain data. Then, they can parse the data and discard what they do not need. To consume the event stream, set up an event listener in your dApp using the following code:
```

File: source/docs/casper/developers/dapps/monitor-and-consume-events.md [hunk 2]
```diff
@@ -1,664 +1,488 @@
...
 
-`DeployProcessed` event is emitted when a given Deploy has been executed. 
+### DeployExpired
 
-The following example contains the JSON representation of the `DeployProcessed` event structure.
+A `DeployExpired` event is emitted when the Deploy is no longer valid for processing or being added to a block due to its time to live (TTL) having expired.
 
 <details>
-<summary>Expand the below section to view the DeployProcessed event details:</summary>
+  <summary>Expand to view output</summary>
```
Comment (ACStoneCL) [inline source/docs/casper/developers/dapps/monitor-and-consume-events.md:251]:
```suggestion
<summary>Expand to view output</summary>
```

File: source/docs/casper/developers/dapps/monitor-and-consume-events.md [hunk 3]
```diff
@@ -1,664 +1,488 @@
...
+* [timestamp](../../concepts/serialization-standard.md#timestamp) - A timestamp representing the moment the validator faulted.
 
-### Fault event
+</details>
 
-The `Fault` event is emitted if there is a validator error. 
+### FinalitySignature
 
-The following example contains the JSON representation of the `Fault` event structure.
+This event indicates that the final approvals from validators are signed, and further alterations to the block will not be allowed. Refer to the [consensus reached](../../concepts/deploy-and-deploy-lifecycle.md#consensus-reached) and [block finality](../../concepts/glossary/B.md#block-finality) sections to learn more about finality signatures.
```
Comment (ACStoneCL) [inline source/docs/casper/developers/dapps/monitor-and-consume-events.md:290]:
```suggestion
This event indicates validators have signed the final approvals and further alterations to the block will not be allowed. Refer to the [consensus reached](../../concepts/deploy-and-deploy-lifecycle.md#consensus-reached) and [block finality](../../concepts/glossary/B.md#block-finality) sections to learn more about finality signatures. 
```

**Review by ACStoneCL — COMMENTED** [review]
A few small nits, but otherwise this LGTM - but with the caveat that @ipopescu mentioned about Sidecar ideally supplanting most direct connections.

Will leave final approval to @hoffmannjan or whoever performs the final tech review.

**Comment by andrzej-casper** [comment]
@ipopescu Build fixed by merging latest `dev`.

**Review by ipopescu — APPROVED** [review]
LGTM!

File: source/docs/casper/developers/dapps/monitor-and-consume-events.md [hunk 4]
```diff
@@ -1,664 +1,488 @@
...
+```javascript
+const { EventStream, EventName } = require("casper-js-sdk")
+
+const es = new EventStream("http://NODE_ADDRESS:9999/events/" + CHANNEL)
+es.start()
+es.subscribe(EventName.EVENT_NAME, eventHandler)
+
+function eventHandler(event) {
+    console.log(event)
+}
```
Comment (hoffmannjan) [inline source/docs/casper/developers/dapps/monitor-and-consume-events.md:38]:
```suggestion
const eventHandler = (event) => {
    console.log(event)
}
```

Comment (hoffmannjan) [reply source/docs/casper/developers/dapps/monitor-and-consume-events.md:38]: lets use arrow function everywhere as this is current standard for ES6

File: source/docs/casper/developers/dapps/monitor-and-consume-events.md [hunk 5]
```diff
@@ -1,664 +1,488 @@
...
+
+<Tabs>
+
+<TabItem value="js" label="JavaScript">
+
+```javascript
+function eventHandler(event) {
+  if (event.body.DeployAccepted.header.account == "012481699f9231e36ecf002675cd7186b48e6a735d10ec1b30f587ca716937752c") {
+    // Perform an action
+  }
```
Comment (hoffmannjan) [inline source/docs/casper/developers/dapps/monitor-and-consume-events.md:407]:
```suggestion
const eventHandler = (event) => {
  if (event.body.DeployAccepted.header.account == "012481699f9231e36ecf002675cd7186b48e6a735d10ec1b30f587ca716937752c") {
    // Perform an action
  }
```

**Review by hoffmannjan — APPROVED** [review]
LGTM, only two nitpicks to use more current syntax :)

**Comment by andrzej-casper** [comment]
@dylanireland Build fixed by merging latest `dev`. PR ready for merge :rocket:!

**Pull request merged**